feat(cartpost) : 장바구니 CRUD 기능 구현

sora0319 opened the pull request:
## 📌 목적
- 장바구니 api의 CRUD의 기능 구현을 목적으로 하였고, Mock data로 확인이 가능하도록 구현하는 것이 추가적인 목표입니다

## ✅ 변경 사항
- 장바구니 아이템 목록에 대한 반환을 합니다. 
- 장바구니 아이템에 대한 상세 내용을 반환합니다.
- CartService에 대한 단위 테스트로 코드에 대한 안정성을 높였습니다
- http 파일을 생성하여 postman이 아니라 코드 내부에서 간단하게 테스트 해볼 수 있도록 구성하였습니다.

## 🧪 테스트 방법
1. cartpost-service.yml 파일을 받아 설정합니다 (h2)
2. 서비스를 실행해주시고, h2-console을 열기 위해 (localhost:{cartpost의 랜덤 포트 번호}/h2-console에 접속합니다
3. 테이블과 데이터가 생성되었는지 확인합니다
4. srt/main/http/CartApi.http를 실행하여 각 api에 대해 통신이 되는지 확인합니다
5. CartServiceTest를 실행하여 단위 테스트가 작동하는지 확인합니다

## 📎 참고 사항
### 카트 아이템 목록
<img width="655" height="596" alt="스크린샷 2025-11-17 065747" src="https://github.com/user-attachments/assets/e69d2302-c2b2-4fb7-b1ff-e701b44d79c0" />

### 카트 아이템 목록(장바구니에 담긴 것이 없을 때)
<img width="483" height="422" alt="스크린샷 2025-11-17 083352" src="https://github.com/user-attachments/assets/18d24758-4605-473c-8aaa-397a7c5b5a10" />

### 카트 아이템 목록 오류(장바구니 생성 안됨)
<img width="810" height="478" alt="스크린샷 2025-11-17 065722" src="https://github.com/user-attachments/assets/8548938a-24dd-4e8c-8db6-576c1df418ff" />

- 카트 아이템 상세

- 카트 아이템 상세  오류(존재하지 않는 아이템)
<img width="539" height="284" alt="스크린샷 2025-11-17 083448" src="https://github.com/user-attachments/assets/5483f953-0e43-4031-b808-f8bba00339d3" />


### H2 console 
<img width="911" height="427" alt="스크린샷 2025-11-17 084109" src="https://github.com/user-attachments/assets/1f04a2d3-7a30-41f5-a333-d0008c3f26df" />



## 📝 제출자 검토 리스트

- [x] 실행 시 오류가 없나요?
- [x] 테스트 전부 통과 했나요?
- [x] 코드 컨벤션을 모두 지켰나요?
- [x] 브랜치가 develop에 merge 요청을 보냈나요?

## 📝 검토자 체크 리스트

> [!IMPORTANT]
> 직접 모든 부분에 대해 검토하고 멘션이나 이모티콘 남겨주세요

- [x] 요청을 받은 후 하루 이내에 피드백을 보내주세요
- [x] 본인이 검토해야하는 PR 피드백을 주세요.
- [x] 브랜치가 develop에 merge 요청을 보냈나요?

sora0319 committed:
feat(cart): 장바구니 모듈 의존성 추가

Lombok, JPA, H2, Swagger 의존성 추가함

sora0319 committed:
feat(cart,post): swagger API 문서 기본 설정

sora0319 committed:
feat(cart,post): 자동 날짜 업데이트 JPA 기능 추가

sora0319 committed:
feat(cart,post): BaseEntity 추가

sora0319 committed:
feat(cart,post): ResponseDto 추가

sora0319 committed:
feat(cart,post): 빈 데이터 값을 위한 EmptyDto

sora0319 committed:
feat(cart): cart controller를 위한 swagger 파일 추상화 생성

sora0319 committed:
chore(cart,post): code convention에 맞추어 수정

sora0319 committed:
feat(cart,post): 자동으로 dto 조회하도록 defaultValue 삭제

sora0319 committed:
feat(cart,post): 예외 처리를 위한 공통 예외 생성

sora0319 committed:
feat(cart,post): 예외 처리를 위한 예외 처리 코드 모음 생성

sora0319 committed:
feat(cart,post): valid 유효성 검사 에러를 관리하기 위한 정보 객체 생성

sora0319 committed:
feat(cart,post): 응답 dto에 에러 처리를 위한 팩토리 메소드 생성

sora0319 committed:
feat(cart,post): 전역 에러 관리 handler 생성

sora0319 committed:
chore(cart,post): 전역 에러 관리 handler return value 이름 수정

sora0319 committed:
chore(cart,post): EmptyDto를 EmptyResponse로 이름 변경

sora0319 committed:
feat(cart,post): swagger schema 추가

sora0319 committed:
chore(cart,post): EmptyDto를 EmptyResponse로 변경

sora0319 committed:
feat(cart): 아이템 목록 반환 response 생성

sora0319 committed:
feat(cart): 장바구니 controller 생성

sora0319 committed:
feat(cart, post): lombok 적용 코드 추가

sora0319 committed:
chore(cart, post): 사용 안하는 import 제거

sora0319 committed:
feat(cart, post): 컬럼 속성 명시 추가

sora0319 committed:
feat(cart): cart entity 추가

sora0319 committed:
feat(cart): cart item entity 추가

sora0319 committed:
feat(cart): cart repository 추가

sora0319 committed:
feat(cart): cart service 추가

sora0319 committed:
feat(cart): cart service 파라미터 변경

sora0319 committed:
feat(cart): cart service response dto 생성

sora0319 committed:
feat(cart): cart service response dto 생성

sora0319 committed:
feat(cart): 타입 record 로 변경

sora0319 committed:
feat(cart): item code 제거

sora0319 committed:
feat(cart): cart code 추가

sora0319 committed:
feat(post): 지불 방식 타입 추가

sora0319 committed:
feat(post): 의뢰글 태그 엔티티 생성

sora0319 committed:
feat(post): 의뢰글 엔티티 생성

sora0319 committed:
feat(post): commissions repository 생성

sora0319 committed:
feat(post): commissions tag repository 생성

sora0319 committed:
chore(cart,post): 날짜 변수명 변경

sora0319 committed:
chore(post): 의뢰글 controller dto 생성

sora0319 committed:
feat(post): 의뢰글 controller dto 생성

sora0319 committed:
chore(post): 의뢰글 service dto 생성

sora0319 committed:
chore(post): 의뢰글 service dto 생성

sora0319 committed:
chore(post): 의뢰글 service dto 생성

sora0319 committed:
chore(post): 의뢰글 service dto 생성

sora0319 committed:
chore(post): 의뢰글 controller 메서드 생성 및 swagger 문서용 분리

sora0319 committed:
chore(post): 의뢰글 controller 메서드 생성 및 swagger 문서용 분리

sora0319 committed:
feat(post): 의뢰글 service 생성

sora0319 committed:
chore(post): 의뢰글 service 생성

sora0319 committed:
feat(post): 의뢰글 service 생성

sora0319 committed:
Merge remote-tracking branch 'origin/feat/cart/7' into feat/cart/7

sora0319 committed:
Merge remote-tracking branch 'origin/feat/cart/7' into feat/cart/7

sora0319 committed:
chore(cart): 파일 이름 복수형 수정

sora0319 committed:
feat(cart): cart service 구현체 생성

sora0319 committed:
chore(cart): cart service 구현체 위치 변경

sora0319 committed:
test(cart): cart service 단위 테스트 기본 틀 구현

sora0319 committed:
test(post): 의뢰글 service 테스트 기본 틀 생성

sora0319 committed:
feat(post): 의뢰글 service 구현체 기본 틀 생성

sora0319 committed:
feat(common): 전체 DDL sql 문 생성

sora0319 committed:
feat(cart,post): 삭제 상태 초기화 및 상태 변경 메서드 추가

sora0319 committed:
refactor(cart,post): ErrorCode를 CustomStatusCode로 이름 변환

기존 ErrorCode로 명명되어 있을 때에는 성공 시 반환하는 코드에 대한 역할이 부자연스러웠습니다. 이러한 문제점을 해결하기 위해 이름을 CustomStatusCode로 변경하여 성공, 실패에 대한 상태 코드를 관리하도록 하였습니다

sora0319 committed:
refactor(cart): uri의 명확성을 위해 수정

sora0319 committed:
fix(cart): 테이블 이름 오류 수정

sora0319 committed:
refactor(post): 기존 파라미터에 페이지 관리 파라미터를 입력 받았던 것을 Pageable로 관리하도록 수정

기존에는 페이지 요청에 따른 파라미터들을 각각 입력 받았지만, Pageable로 변경하면서

sora0319 committed:
fix(post): ID가 Long 기준으로 되어 있는데 code를 기준으로 String으로 등록된 것을 수정

sora0319 committed:
refactor(post): MONTHLY, PER_JOB으로 각 금액 지불 방식 이름 명명 수정

sora0319 committed:
refactor(cart,post): 2개의 서비스에서 모두 사용해서 클래스 위치 변경

sora0319 committed:
refactor(cart): 응답 객체 1개가 아닌 리스트 형을 반환하도록 수정

sora0319 committed:
feat(cart): h2 전용 데이터로 변경

sora0319 committed:
feat(cart,post): 성공한 경우 dto 생성 메서드 생성

sora0319 committed:
feat(cart,post): 빈 데이터에 대해 직렬화를 위한 get 메소드 생성

sora0319 committed:
chore(cart): service에서 servie 전용 dto 제거

service에서 전용 response, request dto를 사용하려고 하였지만 현재 간단한 MVC 구조이기에 복잡성이 높아지고, mapper도 필요하여 service 전용 dto 사용을 철회

sora0319 committed:
feat(cart): member code로 cart 찾는 메서드 추가

sora0319 committed:
feat(cart): cart code로 cart item 목록 찾는 메서드 추가

sora0319 committed:
feat(cart,post): 데이터가 없거나, 내부 오류에 대한 상태 코드 작성

sora0319 committed:
feat(cart): 사용자아게 보여주기 위한 데이터 추가

sora0319 committed:
feat(cart): 장바구니 아이템 목록 전달 메서드 구현

sora0319 committed:
feat(cart): 장바구니 아이템 목록 전달 dto 구현

sora0319 committed:
feat(cart): 장바구니 아이템 목록 전달 controller 구현

sora0319 committed:
feat(cart): 장바구니 아이템 목록 전달 api test 용 http 문서 작성

sora0319 committed:
feat(cart): 장바구니 아이템 누락된 컬럼 추가 및 enum 타입 명시

sora0319 committed:
feat(cart): 장바구니 아이템 코드 전달 데이터 추가

sora0319 committed:
feat(cart): 계약 상태 코드 추가

sora0319 committed:
feat(cart): 아이템 목록 조회 시 계약이 confirmed 된 것만 조회하도록 변경

sora0319 committed:
feat(cart): 아이템 목록 삭제 api 테스트 코드 생성

sora0319 committed:
feat(cart): 아이템, 장바구니 더미 데이터 수정

sora0319 committed:
feat(cart): 소유권이 없는 아이템에 접근 시 오류 코드 생성

sora0319 committed:
feat(cart): 아이템 삭제 controller 구현

sora0319 committed:
feat(cart): 아이템 삭제 service 메소드 구현

sora0319 committed:
feat(cart): 아이템 조회 메소드 구현

sora0319 committed:
feat(cart): 트랜잭션 관리로 어노테이션 추가

sora0319 committed:
feat(cart): 엔티티 생성자 builder 추가

sora0319 committed:
feat(cart): service 기본 예외 상황 및 성공 상황 테스트 코드 작성

sora0319 committed:
Merge branch 'develop' into feat/cartpost/57

# Conflicts:
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/controller/CartApi.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/controller/CartController.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/controller/dto/response/CartItemsGetResponse.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/model/CartItemsEntity.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/model/CartsEntity.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/repository/CartItemsRepository.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/repository/CartsRepository.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/service/CartService.java
#	cartpost-service/src/main/java/com/example/cartpostservice/cart/service/CartServiceImpl.java
#	cartpost-service/src/main/java/com/example/cartpostservice/common/dto/EmptyResponse.java
#	cartpost-service/src/main/java/com/example/cartpostservice/common/dto/ResponseDto.java
#	cartpost-service/src/main/java/com/example/cartpostservice/common/exception/CustomStatusCode.java
#	cartpost-service/src/main/resources/schema.sql
#	cartpost-service/src/test/java/com/example/cartpostservice/cart/service/CartServiceTest.java

ZZAMBAs commented on cartpost-service/src/main/java/com/example/cartpostservice/cart/service/CartServiceImpl.java:
```diff
@@ -1,19 +1,86 @@
+            return Integer.parseInt(amount);
+        }
+
+        return (int) (days * Integer.parseInt(amount));
```
`amount`가 단위 금액이 맞을까요? 단위 금액이 월 단위일 것 같아서 일급 단위로 변환한 후 곱해야 할 것 같아요. 
그리고 금액이 엄청 커질 수도 있으니 `Long`으로 반환하는 것은 어떨까요?

sora0319 replied:
아!! 맞네요
혹시 일급 계산할 때 월급(amount)에서 몇일을 나누셨나요?

ZZAMBAs replied:
저는 그 때 슬랙에 말씀하신 부분 아직은 안 고쳐져 있기는 한데, 30일 단위로 나누어 계산할 것 같습니다.

sora0319 replied:
넵 저도 그러면 그에 맞추어 그렇게 수정하겠습니다

ZZAMBAs commented on cartpost-service/src/test/java/com/example/cartpostservice/cart/service/CartServiceTest.java:
꼼꼼한 테스트 코드인 것 같습니다👍

ZZAMBAs reviewed (approved):
고생하셨습니다!

sora0319 committed:
feat(post): 의뢰글 생성 요청 dto 구현

sora0319 committed:
feat(post): requestbody 속성 추가

sora0319 committed:
feat(post): 서비스 기능 분리를 위한 MangerService 생성

현재 commissions, tag에 대한 엔티티를 관리는 각각의 service에서 해야 한다고 생각을 합니다. 그렇기에 2개의 service들을 관리할 통합 service 단이 필요하여 ManagerService를 생성하였습니다

sora0319 committed:
feat(post): 공통 서비스 행동을 위한 CrudService 생성

commissions, tag entity, repository 들을 각각 관리하는데에 있어서 같은 행동을 할 것이기 때문에 같은 service에서 관리를 해주는 것이 효율적이라고 생각이 들어 인터페이스로 관리

sora0319 committed:
feat(post): 새로운 공통 인터페이스 엔티티 구현체

sora0319 committed:
feat(post): 누락된 속성 값 추가

haeun9988 reviewed (approved):
다른 분이 리뷰해주신 것 이외에 흠잡을 곳이 없는 코드인 것 같습니다 LGTM!

sora0319 committed:
refactor(cart): 단위 금액 계산을 월단위에서 일단위로 변경하고, 근로 날짜에 해당하는 임금을 계산하는 방식으로 변환
